source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/project/settings/with-settings.tsx

```diff
@@ -0,0 +1,59 @@
+import * as React from 'react';
+import { compose } from 'recompose';
+import { withData, WithDataProps } from 'react-orbitjs';
+import { withTranslations, i18nProps } from '@lib/i18n';
+import * as toast from '@lib/toast';
+
+import { TYPE_NAME as PROJECT } from '@data/models/project';
+import { defaultOptions } from '@data';
+
+
+export function withSettings(WrappedComponent) {
+
+  class DataWrapper extends React.Component<WithDataProps & i18nProps> {
+
+    getMessage = (fieldName, type = 'on') => {
+
+      const { t } = this.props;
+
+      return t(`project.operations.${fieldName}.${type}`);
+
+    }
+
+    toggleField = async (projectId, fieldName, newToggleState) => {
+
+      const { updateStore } = this.props;
+
+      try {
+
+        updateStore(t => t.replaceAttribute(
+          { type: PROJECT, id: projectId },
```

Comment:
you'll want to use the actual `project` from `props` here instead of this type/id object.

Reply:
see: data/containers/project/with-data-actions#updateAttribute